### PR TITLE
docs: update build description slightly

### PR DESCRIPTION
**Pull request by henryiii**
## Description

Updating several things that are a bit dated or have new recommendations. Mostly was updating to add the min version of setuptools, as versions before 42 are buggy when used in PEP 517 mode.

## Suggested changelog entry:

Just docs.

### CODE REVIEW

**Comment by YannickJadoul** [comment]
Given that you added this to the v2.6.2 milestone: if we release 2.6.2 soon, is this necessary?

**Comment by henryiii** [comment]
It was not supposed to go to stable...

**Comment by YannickJadoul** [comment]
Aha, yes. That makes quite some difference in the diff. I thought you were just backporting to update our `stable` documentation` :-)

File: docs/compiling.rst [hunk 1]
```diff
@@ -84,10 +84,11 @@ for NumPy's parallel compilation distutils tool is included. Use it like this:
 
 The argument is the name of an environment variable to control the number of
 threads, such as ``NPY_NUM_BUILD_JOBS`` (as used by NumPy), though you can set
-something different if you want. You can also pass ``default=N`` to set the
-default number of threads (0 will take the number of threads available) and
-``max=N``, the maximum number of threads; if you have a large extension you may
-want set this to a memory dependent number.
+something different if you want; CMAKE_BUILD_PARALLEL_LEVEL is another choice
```
Comment (henryiii) [inline docs/compiling.rst:87]:
```suggestion
something different if you want; ``CMAKE_BUILD_PARALLEL_LEVEL`` is another choice
```

**Review by YannickJadoul — APPROVED** [review]
Huh, I thought I had already approved this, sorry! The looks good, as ever; thanks, @henryiii!

**Pull request merged**